components/ui/autocomplete/src/main/java/mozilla/components/ui/autocomplete/InlineAutocompleteEditText.kt

```diff
@@ -263,6 +265,29 @@ open class InlineAutocompleteEditText @JvmOverloads constructor(
         }
     }
 
+    private fun pasteAsPlainText(clipData: ClipData) {
+        var paste = ""
+        for (i in 0 until clipData.itemCount) {
+            val text = clipData.getItemAt(i).coerceToText(context)
+            paste += if (text is Spanned) text.toString() else text
+        }
+        val data = text.substring(0, selectionStart) + paste + text.substring(selectionEnd, text.length)
+        setText(data)
+    }
+
+    override fun onTextContextMenuItem(id: Int): Boolean {
+        if (id == android.R.id.paste) {
+            return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                super.onTextContextMenuItem(android.R.id.pasteAsPlainText)
+            } else {
+                val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+                clipboard.primaryClip?.let { pasteAsPlainText(it) }
+                true
+            }
+        }
+        return super.onTextContextMenuItem(id)
```

Comment:
I wonder if we could cover this with a unit test?

Reply:
I added a unit test for that. Can you please review it to see if it is good or not?

Reply:
I suppose that it's nice to have tests for all cases here:
  - `android.R.id.paste` action on Post-M API;
  - `android.R.id.paste` action on Pre-M API;
  - not `android.R.id.paste` action.